sklearn/ensemble/_base.py

```diff
@@ -285,3 +286,16 @@ def get_params(self, deep=True):
             names mapped to their values.
         """
         return super()._get_params("estimators", deep=deep)
+
+    def _more_tags(self):
+        try:
+            allow_nan = all(
+                _safe_tags(est[1])["allow_nan"] if est[1] != "drop" else True
```

Comment:
I agree this is better than `main`. Unfortunately, our tag system for allowing nans do not work with pipelines.

REF: https://github.com/scikit-learn/scikit-learn/issues/27903